### ISSUE TITLE
Allow to limit maximum occurrences of a specific block-type

### ISSUE THREAD

**Issue by remdan** [sulu/sulu#6125]
# Problem description

At the moment it is not possible to set a limit for a block-type. You can only set the limit (maxOccurs) for all blocks not for one specific type.

Example use case: I also use blocks for certain sections like Heading-Area (Hero with Image, Title, Subtitle, ...).
So I have created a section "heading" white blocks in properties. I have one block-type "hero. It should be possible to add two heros dependent on block-conditions like (segments: winter and summer) und for extra content, a sticker as an example, I have an extra block-type "sticker" and this should be selectable only once.

block-type: "hero" --> max. twice
block-type: "sticker" --> max. once

Second example use case: For a Mainpage I have predefined segments like a block-type. One block-type should be shown only once on the site --> make it not possible for a Content-Creator to define this block-type twice.

# Proposed solution

Make it possible to set limit of use for a block-type.

**Comment by nicolas-joubert** [comment]
Great idea, so minOccurs must be interesting too.

**Comment by mamazu** [comment]
You can set this property now.

**Comment by remdan** [comment]
@mamazu how I can set that now? It's not clear for me and nothing about in the docs!

**Comment by mamazu** [comment]
@remdan sorry I didn't properly understand the issue. As you described you can only provide a limit over all block types. The per block type limits don't exist. Sorry. But thanks for poking me on that. That sounds like an interesting thing to try and implement.

**Comment by remdan** [comment]
and it should work for a global and local block too


```
<type name="text_block" minOccurs="1">
```


```
<type ref="text_block" minOccurs="1" maxOccurs="1" />
```

**Comment by mamazu** [comment]
Yes, that's exactly what I was thinking. But this would then have to validate the block settings. Like what if the `minOccurs` inside the blocks is 1 and the `minOccurs` outside is 2 or something.

**Comment by bxschmidt** [comment]
Great idea. In addition it would be nice to have some form of scripting involved similar to visibleCondition/disabledCondition.
